pkg/detectors/falsepositives.go

```diff
@@ -199,3 +209,110 @@ func FilterKnownFalsePositives(ctx context.Context, detector Detector, results [
 
 	return filteredResults
 }
+
+// FilterAllowlistedSecrets filters out results that match allowlisted secrets.
+// This allows users to specify known safe secrets that should not be reported.
+// Supports regex patterns.
+func FilterAllowlistedSecrets(ctx context.Context, results []Result, allowlistedSecrets map[string]struct{}) []Result {
+	if len(allowlistedSecrets) == 0 {
+		return results
+	}
+
+	var filteredResults []Result
+	for _, result := range results {
```

Comment:
I think this might be a good candidate for [`slices.DeleteFunc`](https://pkg.go.dev/slices#DeleteFunc). It inverts the logic a little, but you don't need any loop management so in that sense it's cleaner.